src/server/routes/contract/extensions/accountFactory/write/createAccount.ts

```diff
@@ -63,37 +69,53 @@ export const createAccount = async (fastify: FastifyInstance) => {
       const { simulateTx } = request.query;
       const { adminAddress, extraData, txOverrides } = request.body;
       const {
-        "x-backend-wallet-address": walletAddress,
+        "x-backend-wallet-address": fromAddress,
         "x-account-address": accountAddress,
+        "x-account-factory-address": accountFactoryAddress,
+        "x-account-salt": accountSalt,
         "x-idempotency-key": idempotencyKey,
       } = request.headers as Static<typeof walletWithAAHeaderSchema>;
       const chainId = await getChainIdFromChain(chain);
 
-      const contract = await getContract({
+      const factoryContract = await getContractV5({
         chainId,
-        walletAddress,
         contractAddress,
-        accountAddress,
       });
-      const tx = await contract.accountFactory.createAccount.prepare(
+
+      const deployedAddress = await predictAddress({
+        factoryContract,
         adminAddress,
-        extraData,
-      );
-      const deployedAddress =
-        (await contract.accountFactory.predictAccountAddress(
-          adminAddress,
-          extraData,
-        )) as Address;
+        accountSalt: extraData,
+      });
 
-      const queueId = await queueTx({
-        tx,
-        chainId,
-        simulateTx,
-        extension: "account-factory",
-        deployedContractAddress: deployedAddress,
-        deployedContractType: "account",
-        idempotencyKey,
+      // if extraData is not a hex string, convert it to a hex string
+      // this is the same transformation that is done in the SDK
+      // for predictAddress and createAndSignUserOp
+      // but needed here because we're calling the raw autogenerated abi function
+      const accountSaltHex =
+        extraData && isHex(extraData)
+          ? extraData
+          : stringToHex(extraData ?? "");
+
```

Comment:
I see `data` is required in `factoryCreateAccount` so you're hexifying `""` => `0x` if extraData is not provided.

Reply:
Also if we expect to need this logic in all methods, I wonder if we should write a `parseAccountAbstractionHeaders()` function that returns the AA headers with "hexified" accountSalt.

Reply:
its actually only in this case that we need this, everywhere else i do it for you inside the SDK. But here we're calling the 'raw' abi method for createAccount which does not do any pre processing, so i have to do it before calling that raw abi method.

considering making an extension just to do this lil part for you internally